03-JavaServiceSkeleton/MessageProducerApplication/app/build.gradle

```diff
@@ -0,0 +1,34 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 28
+    buildToolsVersion "29.0.3"
+
+    defaultConfig {
+        applicationId "com.example.shevchenko.message"
```

Comment:
Did you manage to install 2 different apk with the same applicationId? (Producer, Consumer)

Reply:
Thank you! I didn't notice it and this was the reason why I could run only Consumer or Producer app. I have removed the static keyword for the message storage variable in the service as it doesn't need anymore. I can save the state in the String field now.